man/mlvpn.1
.\" generated with Ronn/v0.7.3
.\" http://github.com/rtomayko/ronn/tree/0.7.3
.
.TH "MLVPN" "1" "November 2015" "" ""
.
.SH "NAME"
\fBmlvpn\fR \- MultiLink Virtual Public Network aggregator
.
.SH "SYNOPSIS"
\fBmlvpn\fR \-\-user mlvpn \-\-config /path/to/mlvpn\.conf
.
.SH "DESCRIPTION"
\fBmlvpn\fR is designed to help building aggregation tunnels between two hosts\.
.
.P
Main features are:
.
.IP "\(bu" 4
Bandwidth aggregation
.
.IP "\(bu" 4
High availability (detects link up/down)
.
.IP "\(bu" 4
Security (Privilege separation, auth and cipher using libsodium)
.
.IP "\(bu" 4
Scritable
.
.IP "\(bu" 4
External program control (using unix socket or TCP/HTTP socket)
.
.IP "" 0
.
.P
\fBmlvpn\fR uses privilege seperation to increase security\. The monitor process is running as root, doing only privileged operations, the other process is doing most of the stuff\.
.
.P
In order to be able to aggregate multiple links with mismatched bandwidth, \fBmlvpn\fR uses a weight round\-robin algorithm\. \fBmlvpn\fR then sends packets, according to the capacity of the link\. It should be noted that \fBmlvpn\fR only knows about what it sends, so you should ONLY use \fBmlvpn\fR on the links if you want good balancing of the links\.
.
.SH "OPTIONS"
.
.TP
\fB\-\-config\fR, \fB\-c\fR \fIfile\fR
path to the configuration file
.
.TP
\fB\-\-user\fR, \fB\-u\fR \fIusername\fR
drop privileges to \fIusername\fR
.
.TP
\fB\-\-natural\-title\fR
do not use setproctitle(3)
.
.TP
\fB\-\-name\fR, \fB\-n\fR \fIname\fR
change process title to \fIname\fR (privileged process will be named \fIname\fR [priv]: this option is disabled with \fB\-\-natural\-title\fR is used
.
.TP
\fB\-\-debug\fR
log to stderr instead of syslog
.
.TP
\fB\-D\fR\fItoken\fR
only show messages of specified \fItoken\fR\. See README\.debug\.md for available tokens\. \-D can be specified multiple times
.
.TP
\fB\-\-verbose\fR, \fB\-v\fR
increases verbosity (repeat multiple times to increase more)
.
.TP
\fB\-\-quiet\fR, \fB\-q\fR
decrease verbosity The verbosity will print usefull messages and errors only\. If \fB\-\-quiet\fR is specified, then only errors will be shown / logged
.
.TP
\fB\-\-yes\-run\-as\-root\fR
don\'t drop privileges, kills privilege separation, do not use unless you are hacking \fBmlvpn\fR
.
.TP
\fB\-\-version\fR
display the current version of \fBmlvpn\fR
.
.TP
\fB\-h\fR
show the help
.
.SH "LOGGING"
mlvpn uses syslog if \fB\-\-debug\fR is not specified\. if \fB\-\-debug\fR is specified, mlvpn sends logging messages to syslog\.
.
.SH "FILES"
.
.TP
\fI/etc/mlvpn/*\.conf\fR
configuration files for \fBmlvpn\fR
.
.SH "DEBUGGING"
refer to README\.debug\.md for more debugging informations\.
.
.SH "SEE ALSO"
mlvpn\.conf(5)
.
.SH "LINKS"
.
.IP "\(bu" 4
https://github\.com/zehome/MLVPN
.
.IP "\(bu" 4
http://mlvpn\.readthedocs\.org/
.
.IP "" 0
.
.SH "AUTHORS"
.
.IP "\(bu" 4
Laurent Coustet Initial work, current maintainer
.
.IP "" 0



debian/changelog
mlvpn (2.3.0~bpo70+1) wheezy-backports; urgency=medium

  *  Update to 2.3.0

 -- Laurent Coustet <laurent.coustet@clarisys.fr>  Mon, 16 Nov 2015 15:03:46 +0100

mlvpn (2.1.4~bpo70+1) wheezy-backports; urgency=medium

  * Update to 2.1.4 release.
  * Rebuild for wheezy-backports.

 -- Laurent Coustet <laurent.coustet@clarisys.fr>  Fri, 17 Apr 2015 11:35:12 +0200

mlvpn (2.1.3~bpo70+2) wheezy-backports; urgency=medium

  * Update to 2.1.3 release.
  * Rebuild for wheezy-backports.

 -- Laurent Coustet <laurent.coustet@clarisys.fr>  Thu, 16 Apr 2015 10:49:44 +0200

mlvpn (2.1.2~bpo70+1) wheezy-backports; urgency=medium

  * Rebuild for wheezy-backports.
  * Rebuild for wheezy-backports.

 -- Laurent Coustet <laurent.coustet@clarisys.fr>  Mon, 30 Mar 2015 13:36:52 +0200

mlvpn (2.1.2) unstable; urgency=medium

  * Update to 2.1.2 release.

 -- Laurent Coustet <ed@zehome.com>  Tue, 17 Mar 2015 13:27:54 +0100

mlvpn (2.1.0) unstable; urgency=medium
  * Update to 2.1.0 release.

 -- Laurent Coustet <ed@zehome.com>  Sun, 08 Mar 2015 20:06:20 +0100

mlvpn (2.0.5) unstable; urgency=low
  * Update to 2.0.5 release.

 -- Laurent Coustet <ed@zehome.com>  Tue, 17 Feb 2015 01:05:24 +0100

mlvpn (2.0.4) unstable; urgency=low
  * Update to 2.0.4 release.

 -- Laurent Coustet <ed@zehome.com>  Mon, 16 Feb 2015 23:25:50 +0100

mlvpn (2.0.3) UNRELEASED; urgency=low
  * Update to 2.0.3 release.

 -- Laurent Coustet <ed@zehome.com>  Wed, 11 Feb 2015 22:57:49 +0100

mlvpn (2.0.2) UNRELEASED; urgency=low
  * Uses libev / libsodium instead of custom event system.

 -- Laurent Coustet <ed@zehome.com>  Mon, 02 Feb 2015 22:51:36 +0100

mlvpn (1.3.2) UNRELEASED; urgency=low
  * Improve debian wheezy support. (Uses /run instead of /var/run)
  * closes #31

 -- Laurent Coustet <ed@zehome.com>  Wed, 12 Feb 2014 20:16:18 +0100

mlvpn (1.3-1) UNRELEASED; urgency=low
  * Buffer system refactoring (Improving performance)
  * Introduce new system to re-order packets on the receiving side.
    (Can be disabled with compile option and configuration)

 -- Laurent Coustet <ed@zehome.com>  Sun, 28 Oct 2012 18:40:00 +0200

mlvpn (1.2-1) UNRELEASED; urgency=low
  * New version, not a public release yet
  * Based on mlvpn 1.2 (dev/untested)

 -- Laurent Coustet <ed@zehome.com>  Sat, 05 May 2012 16:47:19 +0200

mlvpn (1.0-1.1) UNRELEASED; urgency=low

  * Non-maintainer upload.
  * Create debian user for mlvpn before install

 -- Laurent Coustet <laurent.coustet@clarisys.fr>  Fri, 09 Mar 2012 10:47:36 +0100

mlvpn (1.0-1) UNRELEASED; urgency=low

  * Initial release.

 -- Laurent Coustet <laurent.coustet@clarisys.fr>  Fri, 02 Mar 2012 11:38:42 +0100


www/js/mlvpn.js
var url = "http://localhost:1040/status";
var bwchart;
var bwchart_updateTimer;
var bwchart_maxpoints = 100;
var mlvpn_tunnels = [];

function bwchart_setup(container)
{
    bwchart = new Highcharts.Chart({
        chart: {
            renderTo: container,
            type: 'area',
            animation: false,
            height: 350,
        },
        title: {
            text: 'Bandwidth utilization'
        },
        xAxis: {
            type: 'datetime',
            tickPixelInterval: 150
        },
        yAxis: {
            title: {
                text: 'Bytes/s'
            },
        },
        plotOptions: {
            series: {
                stacking: 'normal',
                lineWidth: 1,
                lineColor: '#ffffff',
                marker: {
                    enabled: false
                }
            }
        },
        tooltip: false,
        legend: {
            align: 'left',
            verticalAlign: 'top',
            y: 0,
            x: 50,
            floating: true,
            borderWidth: 0
        },
        credits: {
            enabled: false
        }
    });

    bwchart_updateTimer = setInterval(function() {
        $.getJSON(url,
            bwchart_refresh,
            function(json) {
                alert("Error: "+x);
                stopTimer(bwchart_updateTimer);
            }
        );
    }, 1000);
}

function pretty_print_size(size, factor, display_unit)
{
    var unit = "B";
    var value = size;
    var tsize = size * factor;

    if (tsize > 1024 * 1024 * 1024) {
        unit = "GiB";
        value = size / 1024 / 1024 / 1024;
    } else if (tsize > 1024 * 1024) {
        unit = "MiB";
        value = size / 1024 / 1024;
    } else if (tsize > 1024) {
        unit = "KiB";
        value = size / 1024;
    }
    if (display_unit)
        return Math.round(value) + " " + unit;
    else
        return Math.round(value);
}

function bwchart_refresh(data)
{
    var i, j;
    var series = bwchart.series;
    var now = (new Date()).getTime();
    for (i = 0; i < data.tunnels.length; i++)
    {
        var tun = data.tunnels[i];
        for(j = 0; j < series.length; j++)
        {
            var s = series[j];
            if (s.name == tun.name)
            {
                var shift = s.data.length > bwchart_maxpoints;
                var y = tun["sentbytes"] - s.lastValue;
                if (! y)
                    y = 0;
                s.lastValue = tun["sentbytes"];
                s.addPoint([now,y], false, shift);
                /* Update progress bar */
                var percentage = Math.round(y / tun["bandwidth"] * 100);
                $('#'+tun["name"]+' .bar').attr("style", 'width: '+percentage+'%');
                $('#'+tun["name"]+' .bw_up').html("Upload: <strong>"+pretty_print_size(y, 1.2, true)+"/s</strong>");
                break;
            }
        }
    }
    bwchart.redraw();
}

function initial_dump(json)
{
    console.log(json);

    /* setup chart */
    bwchart_setup('bwchart');

    /* tuntap if display */
    var tuntap = $('#tuntap');
    var container = $('<div>');

    /* All tunnels */
    mlvpn_tunnels = json.tunnels;
    for (var i = 0; i < mlvpn_tunnels.length; i++)
    {
        var tun = mlvpn_tunnels[i];
        var tundiv = $('<div id="'+tun["name"]+'">');
        tundiv.addClass("tun well");

        /* Status button */
        var statusbutton = $('<div class="btn status"><h1>'+tun.name+'</h1></div>')
        if (tun["status"] == "connected")
            statusbutton.addClass("btn-success");
        else if (tun["status"] == "waiting peer")
            statusbutton.addClass("btn-warning");
        else
            statusbutton.addClass("btn-danger");
        tundiv.append(statusbutton);

        /* Some informations */
        var basicinfos = $('<dl class="dl-horizontal">');
        basicinfos.append($('<dt>Mode</dt><dd>'+tun["mode"]+'</dd>'));
        basicinfos.append($('<dt>Encapsulation</dt><dd>'+tun["encap"]+'</dd>'));
        basicinfos.append($('<dt>Direction</dt><dd>'+
            tun["bindaddr"]+':'+tun["bindport"]+
            " <strong>-&gt;</strong> "+
            tun["destaddr"]+':'+tun["destport"]+'</dd>'));
        tundiv.append(basicinfos);

        /* BW Usage if appropriate */
        if (tun["bandwidth"] != "0" || 1)
        {
            var progressbar = $('<div class="progress progress-sucess progress-striped active"></div>');
            progressbar.append($('<div class="bar" style="width: 0%;"></div>'));
            tundiv.append(progressbar);
            tundiv.append($('<span class="bw_up">Upload: 0 Bytes/s</span>'));
            tundiv.append($('<span class="bw_down">Download: 0 Bytes/s</span>'));
        }
        container.append(tundiv);

        /* Chart series */
        bwchart.addSeries({
            type: 'area',
            name: tun["name"],
            lastValue: 0,
            data: (function() {
                    var data = [];
                    var time = (new Date()).getTime();
                    var i;
                    for (i = -bwchart_maxpoints; i < 0; i++) {
                        data.push({
                            x: time + i * 1000,
                            y: null
                        });
                    }
                    return data;
                })()
        });
    }
    tuntap.html(container);
}

$().ready(function()
{
    $.getJSON(url,
        initial_dump,
        function(json) {
            alert("Error: "+x);
        }
    );
});

